language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/toml/ballerinatoml/completion/BallerinaTomlCompletionTest.java

```diff
@@ -0,0 +1,199 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.langserver.toml.ballerinatoml.completion;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+import org.ballerinalang.langserver.commons.workspace.WorkspaceDocumentException;
+import org.ballerinalang.langserver.completion.util.CompletionTestUtil;
+import org.ballerinalang.langserver.util.FileUtils;
+import org.ballerinalang.langserver.util.TestUtil;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.jsonrpc.Endpoint;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Completion Test Interface.
+ *
+ * @2.0.0
+ */
+public abstract class BallerinaTomlCompletionTest {
```

Comment:
Do we have another toml file in ballerina which we can generate completions solely based on schema? if so is it possible to add that implementation and a test case? (if its within the scope of this PR)

Reply:
completion item generation based on schema  is tested for both Ballerina.toml and Clould.toml  in the commons/test/toml/schema_visitor tests.

Reply:
Those are unit tests right? the idea behind my suggestion is to verify the functionality when theres two or more extensions. Ex - Ballerina.toml , Cloud.toml

Reply:
Got your point. To test that we would have to add the specific implementation(Cloud, Dependencies) with this PR.

Reply:
Yup, Correct. we cant add Cloud.toml into this because the implementation is in another repo. If we want a simple implementation we can provide the impl for Settings.toml as the schema is simple. but again, its another task.If this this is beyond the scope, we can always verify the functionality manually for now.